server/src/main/java/org/opensearch/action/search/SearchRequestContext.java

```diff
@@ -144,15 +145,15 @@ public SearchRequest getRequest() {
         return searchRequest;
     }
 
-    void setSuccessfulSearchShardIndices(Set<String> successfulSearchShardIndices) {
+    void setSuccessfulSearchShardIndices(Set<Index> successfulSearchShardIndices) {
         this.successfulSearchShardIndices = successfulSearchShardIndices;
     }
 
     /**
-     * @return A {@link List} of {@link String} representing the names of the indices that were
+     * @return A {@link Set} of {@link Index} representing the names of the indices that were
```

Comment:
This represents the Index objects and not just the names of the indices?

Reply:
Yes, the field data cache will be more precise with the object instead of just name (Map<Index, ...>). Also gives more flexibility to anyone who may use this info in the future.

Reply:
Got it, I meant we could also update the comment if applicable. Thanks